### PR TITLE
Nym::Bip47 Store and Load 

### PR DESCRIPTION
**Pull request by kmels**

### CODE REVIEW

File: src/storage/tree/Nym.cpp [hunk 1]
```diff
@@ -1095,11 +1126,54 @@ bool Nym::Store(
...
+
+        return false;
+    }
+
+    Lock writeLock(write_lock_, std::defer_lock);
+    sLock bip47Lock(bip47_lock_, std::defer_lock);
+    std::lock(writeLock, bip47Lock);
+
+    for (const auto& chain : context.chain()) {
+        const auto it = bip47_contexts_.find(chain.type());
```
Comment (justusranvier) [inline src/storage/tree/Nym.cpp]: auto& chain = bip47_contexts_[chain.type()]

That line will create the inner map if it didn't already exist

File: src/storage/tree/Nym.cpp [hunk 2]
```diff
@@ -1095,11 +1126,54 @@ bool Nym::Store(
...
+
+        if (it == bip47_contexts_.end()) {
+            bip47_contexts_.emplace(
+                chain.type(),
+                std::map<std::string, std::set<Bip47ChannelID>>());
+        }
+
+        for (const auto& channel : chain.channel()) {
+            const auto innerMap = it->second;
```
Comment (justusranvier) [inline src/storage/tree/Nym.cpp]: auto& channels = chain[channel.contact()];
channels.emplace(Bip47ChannelID{channel.contact(), channel.paymentcode()});

File: src/storage/tree/Nym.cpp (hunk 1 above)
Comment (kmels) [reply src/storage/tree/Nym.cpp]: great !! added.

File: src/storage/tree/Nym.cpp (hunk 2 above)
Comment (kmels) [reply src/storage/tree/Nym.cpp]: Thank you, I changed it

File: src/storage/tree/Nym.cpp [hunk 3]
```diff
@@ -1067,6 +1081,25 @@ proto::StorageNym Nym::serialize() const
         *serialized.add_blockchainaccount() = account;
     }
 
+    for (const auto& it_chain : bip47_channels_) {
+        auto& chain = it_chain.first;
+
+        for (const auto& it_contact : it_chain.second) {
+            auto& channel = *serialized.add_bip47channelindex();
+            channel.set_version(1);
```
Comment (justusranvier) [inline src/storage/tree/Nym.cpp]: Use a define here so that the version number can be updated later

**Pull request closed without merging**